tests/pipelines/test_pipelines_text_generation.py

```diff
@@ -212,7 +212,11 @@ def run_pipeline_test(self, text_generator, _):
         # it requires BOS token to exist.
         # Special case for Pegasus which will always append EOS so will
         # work even without BOS.
-        if text_generator.tokenizer.bos_token_id is not None or "Pegasus" in tokenizer.__class__.__name__:
+        if (
+            text_generator.tokenizer.bos_token_id is not None
+            or "Pegasus" in tokenizer.__class__.__name__
+            or "Git" in model.__class__.__name__
+        ):
```

Comment:
@Narsil I had to include this since `GitForCausalLM` doesn't require its BOS token to be set for generation to work. It generates text by starting from `model.config.bos_token_id`

Reply:
Thanks for the ping. This works !